### ISSUE TITLE
host 127.0.0.1:9000: rename /usr/bin/minio /usr/bin/.minio.old: operation not permitted

### ISSUE THREAD

**Issue by pschichtel** [minio/operator#2319]
The chain of operator upgrade issues continues after #2305. This time:

```
main-controller.go:1570] error syncing 'minio/tenant': host 127.0.0.1:9000: rename /usr/bin/minio /usr/bin/.minio.old: operation not permitted
```

When updating the image in the tenant resource.

The upgrade is from `RELEASE.2024-08-29T01-40-52Z` to `RELEASE.2024-09-13T20-26-02Z`.

The rest of the setup is identical to #2305.

**Comment by pschichtel** [comment]
Manually editing the STS "works around" the issue as last time.

**Comment by pschichtel** [comment]
I'm wondering: is my setup somehow special or is the upgrade process via operator just not being used much?

**Comment by harshavardhana** [comment]
It looks like your set-up doesn't allow renames() there is nothing much we can do here. It looks like the image is treated as immutable in your environment.

**Comment by harshavardhana** [comment]
We have tests already for this that work in regular k8s deployment. We don't ever come across any other non k8s deployments so we never really test it.

If you can reproduce on k8s or openshift we can take a look at how we can handle these cases.

Any other vendor here is out of scope for us.

**Comment by harshavardhana** [comment]
Interesting, let me see if I can make that path fully writable for all users.

**Comment by pschichtel** [comment]
@harshavardhana did you also republish the latest images?

**Comment by harshavardhana** [comment]
> @harshavardhana did you also republish the latest images?

I did @pschichtel

**Comment by pschichtel** [comment]
I have this image in production now, let's see how the next upgrade goes :)
